sycl/source/detail/context_impl.cpp

```diff
@@ -332,10 +332,8 @@ std::vector<RT::PiEvent> context_impl::initializeDeviceGlobals(
       // are cleaned up separately from cleaning up the device global USM memory
       // this must retain the event.
       {
-        std::optional<OwnedPiEvent> ZIEvent =
-            DeviceGlobalUSM.getZeroInitEvent(Plugin);
-        if (ZIEvent.has_value())
-          InitEventsRef.push_back(ZIEvent->TransferOwnership());
+        if (OwnedPiEvent ZIEvent = DeviceGlobalUSM.getZeroInitEvent(Plugin))
+          InitEventsRef.push_back(ZIEvent.TransferOwnership());
```

Comment:
nit: I think move-ctor and `std::move` here might be more idiomatic, but this is clear as well.

Reply:
Good point, though I like how this makes it very explicit that the ownership is transferred. If you think move is better I can make a follow-up patch.